src/config/setup.h

```diff
@@ -331,27 +332,16 @@ class PropHex final : public Property {
 
 #define NO_SUCH_PROPERTY "PROP_NOT_EXIST"
 
-typedef void (*SectionFunction)(Section*);
+using SectionUpdateHandler =
+        std::function<void(SectionProp*, const std::string& prop_name)>;
```

Comment:
How about changing the API to use a reference (`const SectionProp&`) instead of a pointer?

Reply:
I think I can do that easily in a new commit. Good idea.